vowpalwabbit/conditional_contextual_bandit.cc

```diff
@@ -455,18 +474,13 @@ void learn_or_predict(ccb& data, multi_learner& base, multi_ex& examples)
     {
       if (should_augment_with_slot_info)
       {
+        data.generated_interactions = data.inter_gen.generated_interactions;
         // Namespace crossing for slot features.
-        data.generated_interactions.clear();
-        {
-          // lock while copying interactions since the parsing thread might be adding more interactions
-          // this should only cause contention when using -q ::
-          std::unique_lock<std::mutex> lock(data.original_interactions->mut);
-          data.generated_interactions.append(*data.original_interactions);
-        }
         calculate_and_insert_interactions(data.shared, data.actions, data.generated_interactions);
-        data.shared->interactions = &data.generated_interactions;
-        for (auto* ex : data.actions) { ex->interactions = &data.generated_interactions; }
       }
+      // Since CCB must interact its own namespaces this operation must always be done to ensure : is expanded.
+      data.shared->interactions = &data.generated_interactions;
+      for (auto* ex : data.actions) { ex->interactions = &data.generated_interactions; }
```

Comment:
It'd be nice having a guard for this, but its not really necessary since all this is just mucking with member objects

Reply:
I'll come back to this as i keep working on this file